### PR TITLE
New Recipe: LibDeflate v0.1.0

### PR DESCRIPTION
**Pull request by jakobnissen**
This is my first JLL package, so I'm not really sure I know what I'm doing. Any feedback would be appreciated.

### CODE REVIEW

File: L/LibDeflate/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "LibDeflate"
+version = v"0.1.0"
```
Comment (aviks) [inline L/LibDeflate/build_tarballs.jl:6]: Version should be the same as the underlying library, `v1.6` in this case.

File: L/LibDeflate/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/ebiggers/libdeflate/archive/v1.6.tar.gz", "60748f3f7b22dae846bc489b22a4f1b75eab052bf403dd8e16c8279f16f5171e")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd libdeflate-1.6/
+if [ ${target} = "x86_64-w64-mingw32" ]; then
```
Comment (aviks) [inline L/LibDeflate/build_tarballs.jl:17]: You can check wildcards here: *-mingw32

File: L/LibDeflate/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    RESULT=$TARGET
+elif [ ${target} = "x86_64-apple-darwin14" ]; then
+    TARGET=libdeflate.dylib;
+    RESULT=libdeflate.0.dylib
+else
+    TARGET=libdeflate.so;
+    RESULT=libdeflate.so.0
+fi
+make DISABLE_ZLIB=true "GCC=O3" ${TARGET}
+mv ${RESULT} /workspace/destdir/
```
Comment (aviks) [inline L/LibDeflate/build_tarballs.jl:31]: Move it to ${libdir}

Comment (aviks) [reply L/LibDeflate/build_tarballs.jl:31]: Actually, you don't have to do this explicitly. you can set PREFIX like so: https://github.com/ebiggers/libdeflate/blob/master/Makefile#L14

**Comment by aviks** [comment]
Try this

```
script = raw"""
cd $WORKSPACE/srcdir
cd libdeflate-1.6/
export PREFIX=${prefix}
export LIBDIR=${libdir}
make DISABLE_ZLIB=true "GCC=O3"
make install
"""
```

File: L/LibDeflate/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+sources = [
+    ArchiveSource("https://github.com/ebiggers/libdeflate/archive/v1.6.tar.gz", "60748f3f7b22dae846bc489b22a4f1b75eab052bf403dd8e16c8279f16f5171e")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd libdeflate-1.6/
+make PREFIX=${prefix} LIBDIR=${libdir} DISABLE_ZLIB=true "GCC=O3"
+make install
```
Comment (thofma) [inline L/LibDeflate/build_tarballs.jl:18]:
```suggestion
make PREFIX=${prefix} install
```

Comment (thofma) [reply L/LibDeflate/build_tarballs.jl:18]: This should fix it.

File: L/LibDeflate/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "libdeflate"
```
Comment (giordano) [inline L/LibDeflate/build_tarballs.jl:5]: This name _must_ match exactly the name of the directory, case included

**Comment by thofma** [comment]
This needs a patch to work on windows, see https://github.com/JuliaPackaging/Yggdrasil/pull/1449

File: L/LibDeflate/build_tarballs.jl (hunk 5 above)
Comment (jakobnissen) [reply L/LibDeflate/build_tarballs.jl:5]: Dear God I am not good at this. Thank you for your help. So it must be called `libdeflate-1.6` (name of the source directory), or `LibDeflate` (name of this JLL)?

**Comment by jakobnissen** [comment]
> This needs a patch to work on windows, see #1449

Cool, it looks like that one passes, so I can just close this PR, I think. Just to be clear, only the shared library is actually needed. The created (and unneeded) executables will not cause problems by e.g. overwriting the system `gzip`, right?

File: L/LibDeflate/build_tarballs.jl (hunk 5 above)
Comment (giordano) [reply L/LibDeflate/build_tarballs.jl:5]: The name of the directory where the script is.  You can either rename the directory `libdeflate` or change back this variable to `LibDeflate`.  Usually we want to use the same name as the upstream package, which seems to be `libdeflate`?

**Comment by thofma** [comment]
No, no problems. They are not even contained in the `_jll`, since we specifiy the products explicitely.

**Comment by jakobnissen** [comment]
Thank you!

**Pull request closed without merging**